src/Cms/Files.php

```diff
@@ -93,6 +96,40 @@ public function changeSort(array $files, int $offset = 0): static
 		return $this;
 	}
 
+	/**
+	 * Deletes the files with the given IDs
+	 * if they exist in the collection
+	 *
+	 * @throws \Kirby\Exception\Exception If not all files could be deleted
+	 */
+	public function delete(array $ids): void
+	{
+		$exceptions = [];
+
+		// delete all pages and collect errors
+		foreach ($ids as $id) {
+			try {
+				$model = $this->get($id);
+
+				if ($model instanceof File === false) {
+					throw new NotFoundException(
+						key: 'file.notFound'
+					);
+				}
+
+				$model->delete();
+			} catch (Throwable $e) {
+				$exceptions[$id] = $e;
+			}
+		}
+
+		if ($exceptions !== []) {
+			throw new Exception(
+				key: 'file.delete.multiple',
+			);
+		}
+	}
```

Comment:
`Discussion:` I of course understand why it is implemented like this currently, but for a core method in this class, it seems a bit unusual to swallow the actual exceptions and just throw this vague one. Not sure there is a better way.

Reply:
All exceptions are now collected and displayed properly in the error dialog …

![Screenshot 2024-12-13 at 17 22 35](https://github.com/user-attachments/assets/ebdc62d8-e2df-4c2c-be5a-62d59dae1ef2)